server/consumer.go

```diff
@@ -3135,7 +3143,11 @@ func nextReqFromMsg(msg []byte) (time.Time, int, int, bool, time.Duration, time.
 		return time.Time{}, 1, 0, false, 0, time.Time{}, nil
 
 	case req[0] == '{':
-		var cr JSApiConsumerGetNextRequest
+		cr := jsGetNextPool.Get().(*JSApiConsumerGetNextRequest)
+		defer func() {
+			*cr = JSApiConsumerGetNextRequest{}
+			jsGetNextPool.Put(cr)
```

Comment:
The pool returns normally a pointer to `JSApiConsumerGetNextRequest` struct, and cr is a struct, not a pointer to it. So is this correct or should it be `&cr`?

Reply:
This is correct, yes, as `cr` is `*JSApiConsumerGetNextRequest`. `&cr` would be a pointer to a pointer.

Reply:
Oh right, `cr` used to be a struct but is now a pointer to the struct. Sorry for the noise.